src/main/java/com/synopsys/integration/jenkins/JenkinsWrapper.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * jenkins-common
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.jenkins;
+
+import static hudson.model.Items.XSTREAM;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.matchers.IdMatcher;
+
+import hudson.security.ACL;
+import hudson.util.VersionNumber;
+import jenkins.model.Jenkins;
+
+public class JenkinsWrapper {
```

Comment:
It's worth considering whether we want JenkinsWrapper to have methods that produce the helper objects. If we do, we probably want to move JenkinsWrapper and the helpers to a new package (`com.synopsys.integration.jenkins.wrapper` perhaps) and protecting the Jenkins API calls -- getVersion, getCredentialsById, etc so the public api changes to just creating the wrapper to create the helpers, which require the wrapper anyway.

Reply:
So I can see doing this for JenkinsVersionHelper.java and SynopsysCredentialsHelper.java.

Since JenkinsProxyHelper.java can work with and without an instance of JenkinsWrapper, would it make sense to include this?

Reply:
As long as it's not the only way to create a JenkinsVersionHelper, that seems fine.